### PR TITLE
Add the Google account reconnection UI and flow

### LINKED ISSUE #486: Handle Google account disconnect and prompt user to reconnect
### Describe the issue:
Even though the server handles the token and renews it with the refresh token. There are still scenarios where a refresh token could be revoked, which results in the Google account no longer being connected.
This can mainly happen in scenarios where the connection to the app is revoked, or possibly a limit is hit. [Full list of reasons](https://blog.timekit.io/google-oauth-invalid-grant-nightmare-and-how-to-fix-it-9f4efaf1da35) why it might no longer be valid.

When this happens, at the moment we just get failed requests to the API, but we don't catch this explicitly to prompt the user to reconnect. Even when we move to the settings page we can't see the proper status and correct the error:

image

### Expected behavior:

We should get a clear error message in the UI that prompts the user to reconnect.

When sending a request we do get the right response:
```
2021-04-15T11:34:27+00:00 DEBUG Automattic\WooCommerce\GoogleListingsAndAds\API\Google\MerchantReport::get_report_data {"statusCode":400,"error":"Bad Request","message":"Site is not connected to Google account for google-mc"}
```

Maybe we need to add a generic handler to pass this on to the UI. In the UI we see the error:
```
Unable to retrieve report data.
```

It seems to work. And I think it might have some details need to confirm:

1. When a user reconnects a no access Google account (above step 4.ii), and if they browse to other pages in this status, would the HTTP status code still being 401? Thus we can bring user back to the reconnection page again.
2. Inherited from above, in this status, whether the frontend requests `DELETE` to the same API `/wc/gla/google/connect` to disconnect the wrong Google account as well when they click on the "Try another Google account" button? Or requests to the new endpoint?

#### 💡  Google account reconnection flow (added on June 8)

When GLA doesn't have valid credentials available for the Google account, the reconnection flow would be:

1. When the client receives any 401 response from APIs, redirect the user to the reconnection page: `/wp-admin/admin.php?page=wc-admin&path=/google/settings&subpath=/reconnect-accounts`.
2. In the reconnection page, fetch account connection status from a new API (#823) `/wc/gla/google/reconnected`. At this moment, the API response should be:
    ```json
    {
      "active": "no",
    }
    ````
    And the UI status should be:
    image
3. The "Connect" button redirects the user to Google's authorization page. After proceeding with authorization, Google redirects the user back to the reconnection page.
4. The reconnection page requests to account connection status from the API `/wc/gla/google/reconnected` again to get the accessibility of currently connected Google MC and/or Ads accounts.
    1. No: go to step 5.
    2. Yes: go to step 6.
5. User connected to a Google account that doesn't have accessibility of current MC/Ads. The API response should be:
    ```json
    {
      "active": "yes",
      "email": "test@gmail.com",
      "merchant_account": 1234567890,
      "merchant_access": "no",
      "ads_account": 5678901234,
      "ads_access": "no"
    }
    ```
    Adn display the fault-tolerant process UI:
    image
    1. When clicking on "Disconnect all accounts ", popup the confirm modal. Redirects to the Get Started page If the user proceeds with all accounts disconnecting process.
    2. When clicking on "Try another Google account", disconnect Google account and go to step 2 again.
6. Reconnected successfully. Redirect to the dashboard page.

### Steps to reproduce:
1. Connect all the accounts
2. Use the connection test page to disconnect only the Google account
3. Go back to the UI and notice that requests fail and the settings page is not viewable.

### PR DESCRIPTION
**Pull request by eason9487**
### Changes proposed in this Pull Request:

Implemented the rest of #486 and based on #834.

Please refer to the **Google account reconnection flow** section in the #486 description.

- Add a new shared component AccountCard for the new design
- Add selector/resolver/action/reducer to get Google access status from API and to disconnect Google account
- Implement the Google account reconnection UI and flow
- Redirect to the reconnection page from the Setting page if the Google account is disconnected

#### 💡 Additional note

Although the improvement of onboarding process design is still under discussion, the new design for the account card UI has already been included in this PR, so the component was implemented in a way that will continue to be developed in the future. 

#### 🎥 . Reconnect to a Google account with access permissions
https://user-images.githubusercontent.com/17420811/123742036-a571c200-d8dd-11eb-8e63-c924dcb5971f.mp4

#### 🎥 . Reconnect to a Google account that does not have access permissions
https://user-images.githubusercontent.com/17420811/123742249-ed90e480-d8dd-11eb-8c89-0315d27f32ee.mp4

### Detailed test instructions:

1. Disconnect Google account from the Connection Test page
2. Go to the dashboard page
3. When receiving any 401 response error or detecting the Google account being disconnected, the browser should be redirected to the reconnection page `/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings`
4. Proceed with next steps according to the **Google account reconnection flow** section in the #486 description
5. See if the implementation of reconnection flow is working properly as that description

####❗ An edge case

When browsing to the Product Feed page with a disconnected Google account, the `/wc/gla/mc/product-statistics` API returns 400 response error, so it'd not be redirected to the reconnection page. I was thinking if users would be redirected to reconnect by other pages eventually, maybe we could leave it as is.

### Changelog entry

> Add - Google account reconnection UI and flow.

### CODE REVIEW

**Comment by eason9487** [comment]
image

Hey @j111q, when redirecting to the reconnection page, there're a bunch of error messages at the bottom. I thought it could tell users that something went wrong obviously, so I didn't prevent it, it's a little scary to users though.

In addition, since we haven't implemented a more detailed error handling yet. So if we want to avoid it, another option would be hiding all error messages in the short term. What do you think if we leave it as is? And let me know if there are any concerns or ideas. 😄

**Comment by j111q** [comment]
> In addition, since we haven't implemented a more detailed error handling yet. So if we want to avoid it, another option would be hiding all error messages in the short term. What do you think if we leave it as is? 

Thanks for highlighting this @eason9487 ! Yeah I think I'd prefer if we hide all the error messages for this page in the short term. They aren't helpful since they aren't specific enough, and like you said, they're a bit scary. 

I think the Re-connection page makes it clear enough what the intended user action should be - to connect the Google account. Those error messages might confuse or distract from that.

If we do implement more specific error handling, we can say something like `Error: Your Google account has been disconnected. Please connect your Google account again.` But otherwise, let's hide all those generic errors that are there now.

**Comment by eason9487** [comment]
> let's hide all those generic errors that are there now.

@j111q, thanks for replying! 👍  Got it and no problem! I will hide those error messages by another PR.

**Review by j111q — APPROVED** [review]

**Comment by eason9487** [comment]
test slack subscription

File: js/src/components/account-card/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		Logo: GoogleLogo,
+		title: __( 'Google account', 'google-listings-and-ads' ),
+	},
+};
+
+/**
+ * Renders a Card component with account info and status.
+ *
+ * @param {Object} props React props.
+ * @param {string} props.appearance Predefined account enum to indicate the card appearance.
```
Comment (tomalec) [inline js/src/components/account-card/index.js]: :nail_care: 
You can define enumtype in JSDoc

```suggestion
/**
 * Enum of account card appearances.
 *
 * @enum {{Logo: JSX.Element, title: string}}
 */
const appearanceDict = {
	[ GOOGLE ]: {
		Logo: GoogleLogo,
		title: __( 'Google account', 'google-listings-and-ads' ),
	},
};

/**
 * Renders a Card component with account info and status.
 *
 * @param {Object} props React props.
 * @param {appearanceDict} props.appearance Kind of account to indicate the card appearance.
```

Comment (eason9487) [reply js/src/components/account-card/index.js]: The `appearance` prop is string type, but the suggested change seems doesn't fit with it. I adjusted the doc by https://github.com/woocommerce/google-listings-and-ads/pull/838/commits/a52920e91605c8a926daabf50dbd1a7f2f6a12c1.

**Review by tomalec — APPROVED** [review]
Tested locally, works great. Reviewed the code briefly. I'll continue a more detailed review later.
LGTM, ready for the release on Thursday this week.
More cosmetic changes/improvements could be introduced as follow-up.

**Pull request merged**